File: emp_ui/lib/models/employee.dart
class EmployeeModel {
  final int? id;
  final String fullName;
  final String email;
  final String phone;
  final String gender;
  final String employeeType;
  final double salary;
  final String status;
  final int departmentId;
  final String? departmentName;
  final int designationId;
  final String? designationName;
  final String? dateOfJoining;

  EmployeeModel({
    this.id,
    required this.fullName,
    required this.email,
    required this.phone,
    required this.gender,
    required this.employeeType,
    required this.salary,
    this.status = 'ACTIVE',
    required this.departmentId,
    this.departmentName,
    required this.designationId,
    this.designationName,
    this.dateOfJoining,
  });

  factory EmployeeModel.fromJson(Map<String, dynamic> json) {
    return EmployeeModel(
      id: json['id'],
      fullName: json['fullName'],
      email: json['email'],
      phone: json['phone'],
      gender: json['gender'],
      employeeType: json['employeeType'],
      salary: (json['salary'] as num).toDouble(),
      status: json['status'] ?? 'ACTIVE',
      departmentId: json['department'] != null ? json['department']['id'] : 0,
      departmentName: json['department'] != null ? json['department']['name'] : null,
      designationId: json['designation'] != null ? json['designation']['id'] : 0,
      designationName: json['designation'] != null ? json['designation']['name'] : null,
      dateOfJoining: json['dateOfJoining'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "fullName": fullName,
      "email": email,
      "phone": phone,
      "gender": gender,
      "employeeType": employeeType,
      "salary": salary,
      "status": status,
      "department": {"id": departmentId},
      "designation": {"id": designationId},
    };
  }
}

File: emp-api/src/main/java/com/example/entities/Employee.java
package com.example.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;

import java.util.Date;
import java.util.List;

@Entity
@Table(name = "employees_pp")
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String fullName;
    private String email;
    private String phone;
    private String gender;
    private String employeeType;
    private double salary;
    private String status; // ACTIVE / INACTIVE
    @JsonIgnoreProperties({"employees"})
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;
    @JsonIgnoreProperties({"employees"})
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "designation_id", nullable = false)
    private Designation designation;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "date_of_joining", nullable = false, updatable = false)
    private Date dateOfJoining;

    @PrePersist
    protected void onJoin() {
        this.dateOfJoining = new Date();
    }

    // ===== Getters & Setters =====

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getFullName() { return fullName; }
    public void setFullName(String fullName) { this.fullName = fullName; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }

    public String getGender() { return gender; }
    public void setGender(String gender) { this.gender = gender; }

    public String getEmployeeType() { return employeeType; }
    public void setEmployeeType(String employeeType) { this.employeeType = employeeType; }

    public double getSalary() { return salary; }
    public void setSalary(double salary) { this.salary = salary; }

    public Department getDepartment() { return department; }
    public void setDepartment(Department department) { this.department = department; }

    public Designation getDesignation() { return designation; }
    public void setDesignation(Designation designation) { this.designation = designation; }

    public Date getDateOfJoining() { return dateOfJoining; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }


}



